ion/integration_test.go

```diff
@@ -0,0 +1,807 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+const goodPath = "../ion-tests/iontestdata/good"
+const badPath = "../ion-tests/iontestdata/bad"
+const equivsPath = "../ion-tests/iontestdata/good/equivs"
+const nonEquivsPath = "../ion-tests/iontestdata/good/non-equivs"
+
+type testingFunc func(t *testing.T, path string)
+
+type item struct {
+	ionType     Type
+	annotations []string
+	value       []interface{}
```

Comment:
Should `value` be `[]item` to restrict it a bit?

Reply:
We cannot. We are adding different types of Ion in an array of this struct in ```readCurrentValue()```, for example ```reader.StringValue()``` or ```reader.TimeValue()``` and they are of type interface{}

Reply:
Ah, ok. Thanks for explaining.